From: =?utf-8?q?Jonas_=C3=85dahl?= <jadahl@gmail.com>
Date: Thu, 28 Jan 2021 15:04:11 +0100
Subject: backend: Only create idle monitors for added physical input devices

The rest of the things we do aren't applicable, e.g. mapping tablet
devices/tools to monitors and hiding cursors.

Part-of: <https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/1688>
---
 src/backends/meta-backend.c | 19 ++++++++++---------
 1 file changed, 10 insertions(+), 9 deletions(-)

diff --git a/src/backends/meta-backend.c b/src/backends/meta-backend.c
index 4edc366..a862918 100644
--- a/src/backends/meta-backend.c
+++ b/src/backends/meta-backend.c
@@ -397,13 +397,6 @@ meta_backend_monitor_device (MetaBackend        *backend,
   create_device_monitor (backend, device);
 }
 
-static inline gboolean
-device_is_physical_touchscreen (ClutterInputDevice *device)
-{
-  return (clutter_input_device_get_device_mode (device) != CLUTTER_INPUT_MODE_LOGICAL &&
-          clutter_input_device_get_device_type (device) == CLUTTER_TOUCHSCREEN_DEVICE);
-}
-
 static inline gboolean
 check_has_pointing_device (ClutterSeat *seat)
 {
@@ -467,11 +460,15 @@ on_device_added (ClutterSeat        *seat,
 
   create_device_monitor (backend, device);
 
-  if (device_is_physical_touchscreen (device))
-    meta_cursor_tracker_set_pointer_visible (priv->cursor_tracker, FALSE);
+  if (clutter_input_device_get_device_mode (device) ==
+      CLUTTER_INPUT_MODE_LOGICAL)
+    return;
 
   device_type = clutter_input_device_get_device_type (device);
 
+  if (device_type == CLUTTER_TOUCHSCREEN_DEVICE)
+    meta_cursor_tracker_set_pointer_visible (priv->cursor_tracker, FALSE);
+
   if (device_type == CLUTTER_TOUCHSCREEN_DEVICE ||
       device_type == CLUTTER_TABLET_DEVICE ||
       device_type == CLUTTER_PEN_DEVICE ||
@@ -491,6 +488,10 @@ on_device_removed (ClutterSeat        *seat,
 
   destroy_device_monitor (backend, device);
 
+  if (clutter_input_device_get_device_mode (device) ==
+      CLUTTER_INPUT_MODE_LOGICAL)
+    return;
+
   meta_input_mapper_remove_device (priv->input_mapper, device);
 
   /* If the device the user last interacted goes away, check again pointer
